src/controller.ts

```diff
@@ -110,9 +110,11 @@ export async function acceptExCmd(exstr: string): Promise<any> {
 }
 
 import { activeTabId } from "./lib/webext"
-browser.webNavigation.onBeforeNavigate.addListener(async function(details) {
-    if (details.frameId === 0 && details.tabId === (await activeTabId())) {
-        state.mode = "normal"
-    }
-})
+if (browser.webNavigation) {
```

Comment:
Could we have a proxy for browser that handled this stuff gracefully? e.g. `triBrowser` that prevents us from having to cover everything in ifs / care about whether a permission exists.

Reply:
I thought about this but the proxy would need to be async (but we could work around that by prefetching and caching permissions) and another problem would be stopping code from executing.
If we throw, Tridactyl will crash just like when there's no proxy. If instead we return a recursive proxy we'll in the best cases crash just like when there's no proxy and in the worst cases silently fail.